src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -130,20 +166,17 @@ private BoundDeconstructionAssignmentOperator BindDeconstructionAssignment(CShar
             deconstructionSteps.Add(step);
 
             // outputs will either need a conversion step and assignment step, or if they are nested variables, they will need further deconstruction
-            return DeconstructOrAssignOutputs(step, variables, syntax, diagnostics, deconstructionSteps, assignmentSteps);
+            return DeconstructOrAssignOutputs(step, variables, syntax, diagnostics, deconstructionSteps, conversionSteps, assignmentSteps, constructionStepsOpt);
         }
 
         /// <summary>
-        /// This will generate and stack appropriate deconstruction and assignment steps for a tuple type.
-        /// The produced deconstruction step has no Deconstruct method since the tuple already has distinct elements.
+        /// The produces a deconstruction step with no Deconstruct method since the tuple already has distinct elements.
         /// </summary>
         private static BoundDeconstructionDeconstructStep MakeTupleDeconstructStep(
                                                         BoundDeconstructValuePlaceholder targetPlaceholder,
                                                         CSharpSyntaxNode syntax,
                                                         DiagnosticBag diagnostics,
-                                                        ArrayBuilder<DeconstructionVariable> variables,
-                                                        ArrayBuilder<BoundDeconstructionDeconstructStep> deconstructionSteps,
-                                                        ArrayBuilder<BoundDeconstructionAssignmentStep> assignmentSteps)
```

Comment:
Note: those last two parameters were un-used. Removed them and similar ones in next method.